### PR TITLE
feat(store-ui): Create Badge component

### LINKED ISSUE #832: Badge

### PR DESCRIPTION
**Pull request by victorhmp**
## What's the purpose of this pull request?

This PR adds the **Badge** component to our store-ui library.

Closes #832.

## How it works? 

The Badge component enables users to create their own discount (or any type of highlighted property) badges and add them to their UIs.

Here's an example on it being used to display a discount badge on top of some ProductSummary components:

Screen Shot 2021-08-23 at 11 46 12

## How to test it?

Check the deploy preview below. You should find this under `Atoms/Badge`.

### CODE REVIEW

**Comment by victorhmp** [comment]
Checked for stores using the current `Badge` from `theme-ui` exported by `@vtex/store-ui` and didn't find any. So the breaking change introduced in this PR doesn't require extra PRs 😄

**Review by emersonlaurentino — APPROVED** [review]

File: packages/store-ui/src/atoms/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import type { ReactNode, ForwardedRef } from 'react'
+import React, { forwardRef } from 'react'
+
+export interface BadgeProps {
+  testId?: string
+  children?: ReactNode
```
[the remark this answers is not in the document]
Comment (igorbrasileiro) [reply packages/store-ui/src/atoms/Badge/Badge.tsx:6]: Also, i think would be a better idea to use the PropsWithChildren, what do you think about it?

```suggestion
  /**
    * ....
    */
  testId?: string
```

```tsx
  function Badge({ ... }: PropsWithChildren<BadgeProps>, ref)
```

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Nice job @victorhmp 👏 !! Left some comments

File: packages/store-ui/src/atoms/Badge/Badge.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (igorbrasileiro) [reply packages/store-ui/src/atoms/Badge/Badge.tsx:6]: Also props isn't appearing on the Props table on storybook
image

**Review by igorbrasileiro — APPROVED** [review]

File: packages/store-ui/src/atoms/Badge/Badge.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ export interface BadgeProps {
   children?: ReactNode
 }
 
-function Badge(
+const Badge = forwardRef<HTMLDivElement, BadgeProps>(function Badge(
   { testId = 'store-badge', children }: BadgeProps,
   ref: ForwardedRef<HTMLDivElement>
```
Comment (emersonlaurentino) [inline packages/store-ui/src/atoms/Badge/Badge.tsx:14]: if you use `forwardRef` directly on function, you don't need to use type to ref.

```suggestion
  ref
```

**Pull request merged**